rasterio/warp.py

```diff
@@ -241,6 +244,14 @@ def reproject(
         **kwargs)
 
 
+def res_tuple(res):
+    try:
```

Comment:
Just plopping the function at the top level is kinda sloppy, confuses the public API for sure. I'll fix that, either inline or putting it in the right place :+1: